pkg/infrastructure/clusterapi/clusterapi.go

```diff
@@ -158,18 +160,25 @@ func (i *InfraProvider) Provision(ctx context.Context, dir string, parents asset
 	i.appliedManifests = []client.Object{}
 
 	// Create the infra manifests.
+	logrus.Info("Creating infra manifests...")
 	for _, m := range infraManifests {
 		m.SetNamespace(capiutils.Namespace)
 		if err := cl.Create(ctx, m); err != nil {
 			return fileList, fmt.Errorf("failed to create infrastructure manifest: %w", err)
 		}
 		i.appliedManifests = append(i.appliedManifests, m)
+		if updateClusterIDs && fmt.Sprintf("%+T", m) == "*v1beta1.Cluster" {
```

Comment:
I'd prefer if we used an interface check here.

Reply:
Sounds good.  I'll switch this out.  Thanks!

Reply:
Oh yeah, now i remember why i did this.  So we are using client.Object for the data wrapper.  Every time I tried to get kind/type, I got empty back.  Another way I was thinking of solving this is higher up, I can just check for the cluster file and get name from it at that time.  I'll mock that up as well.